src/World.cpp

```diff
@@ -1130,19 +1130,36 @@ void cWorld::TickMobs(std::chrono::milliseconds a_Dt)
 		}  // for i - AllFamilies[]
 	}  // if (Spawning enabled)
 
-	// move close mobs
-	cMobProximityCounter::sIterablePair allCloseEnoughToMoveMobs = MobCensus.GetProximityCounter().getMobWithinThosesDistances(-1, 64 * 16);// MG TODO : deal with this magic number (the 16 is the size of a block)
-	for (cMobProximityCounter::tDistanceToMonster::const_iterator itr = allCloseEnoughToMoveMobs.m_Begin; itr != allCloseEnoughToMoveMobs.m_End; ++itr)
+	class cCallback : public cEntityCallback
 	{
-		itr->second.m_Monster.Tick(a_Dt, itr->second.m_Chunk);
-	}
+		virtual bool Item(cEntity * a_Entity) override
+		{
+			if (!a_Entity->IsMob())
+			{
+				return false;
+			}
+
+			cMonster * Monster = static_cast<cMonster*>(a_Entity);
+
+			// Tick close mobs
+			if (Monster->GetParentChunk()->HasAnyClients())
```

Comment:
As discussed earlier, I still believe this is the cleanest approach, it's just that:
A: I want to be completely sure it's all good, because I slightly changed the way parentChunk is used mid-PR.
B: I am seeing potential redundancy here and I'd like to investigate if `isDestroyed` and `GetParentChunk` can be combined.